### PR TITLE
Add dactyl manuform 4x5 shield

### PR DESCRIPTION
**Pull request by nathanielks**
Hello, hello!

I've added a default keymap for Dactyl Manuform 4x5 keebs, which should make it easier to add additional Dactyl's in the future. I used the [default config in the QMK repo](https://github.com/qmk/qmk_firmware/tree/master/keyboards/handwired/dactyl_manuform/4x5) for consistency.

### CODE REVIEW

File: docs/static/setup.sh [hunk 1]
```diff
@@ -121,6 +121,7 @@ select opt in "${options[@]}" "Quit"; do
     19 ) shield_title="Eek!" shield="eek"; split="n" break;;
     20 ) shield_title="BFO-9000" shield="bfo9000"; split="y"; break;;
     21 ) shield_title="Helix" shield="helix"; split="y"; break;;
+    21 ) shield_title="Dactyl Manuform 4x5" shield="dactyl_manuform_4x5"; split="y"; break;;
```
Comment (CrossR) [inline docs/static/setup.sh:124]:
```suggestion
    22 ) shield_title="Dactyl Manuform 4x5" shield="dactyl_manuform_4x5"; split="y"; break;;
```

File: docs/static/setup.ps1 [hunk 2]
```diff
@@ -58,11 +58,11 @@ catch [System.Management.Automation.CommandNotFoundException] {
...
 Test-Git-Config -Option "user.email" -ErrMsg "Git email not set!`nRun: git config --global user.email 'example@myemail.com'"
 
-$permission = (Get-Acl $pwd).Access | 
+$permission = (Get-Acl $pwd).Access |
 ?{$_.IdentityReference -match $env:UserName `
 	-and $_.FileSystemRights -match "FullControl" `
-		-or $_.FileSystemRights -match "Write"	} | 
-			
+		-or $_.FileSystemRights -match "Write"	} |
+
```
Comment (MvEerd) [inline docs/static/setup.ps1:65]: There are only whitespace changes in this hunk

File: docs/static/setup.sh (hunk 1 above)
Comment (nathanielks) [reply docs/static/setup.sh]: Aha, good eyes, thank you! I've committed your suggestion 😊

**Comment by DuMaM** [comment]
sorry to say this but i thing your work will be rejected :(

Take look here
https://github.com/zmkfirmware/zmk/pull/771

**Comment by petejohanson** [comment]
Thank you, contributor, for your patience with how long review and merge of boards/shields has taken!

There are three recent refactors/changes to boards and shields that require some attention, and then we can finally get this PR merged!

1. Hardware Metadata
1. Pro Micro shield DT naming changes
1. Split changes for BLE advertising

# Hardware Metadata

## The Problem

When first developing the process around contributing new shields/boards to ZMK, we failed to recognize that several key files (setup scripts, documentation page of supported hardware, and GH Action `build.yml` file) required changes, often in the same spot, for every PR. This resulted in immediate merge conflicts for every *other* PR after one was merged, which is a  headache for contributors.

## The Fix

By adding discrete metadata files that are located with the boards/shields in question, and using that metadata to generate setup scripts, website hardware list, etc., users can contributing new hardware descriptions without the need to change the same files that other contributors are changing.

## Next Steps

First, refer to https://zmk.dev/docs/development/hardware-metadata-files to familiarize yourself with the new metadata file format.

Next, you have two options for fixing up your PR:

1. If comfortable with `git rebase`, perform an interactive rebase and revert any changes to `build.yml`, `hardware.md`, or the setup scripts `setup.sh`/`setup.ps1`, and then add the new metadata YAML file. Then force push your branch. Or,
2. Create a new branch from an up-to-date `main`, copy in the files for your new hardware, add the metadata file, then commit and push the new branch. Then, edit your open PR to point to your new branch.

# Pro Micro shield DT naming changes

In #876, we have simplified the DT naming for the "nexus node" we expose for pro-micro compatible boards, deprecating the use of `pro_micro_a`, and renaming `pro_micro_d` node to simply `pro_micro`. For pro-micro boards and shields, you'll need to adjust your DT to use the proper names.

Please see https://zmk.dev/docs/development/new-shield#shield-overlays for the updated docs on this.

# Split Shield Advertising Changes

In addition, if this is a split PR, please see #658 where we have changed our conventions to *remove* the the name from the right sides, to prevent users attempting to pair with them and causing split sync issues. This also includes removing the " Left" suffix from the naming on the left side. See the changes in that PR for examples of what to change with your split shield.

# Getting Help

If you have any questions about any of these changes, please comment here and tag @zmkfirmware/boards-shields or ask in the `#boards-shields` Discord channel.

**Comment by Nicell** [comment]
Hello! We're closing this PR due to there being no activity since the metadata update to ZMK in September of last year. If you are still interested in updating this PR, please let us know, and we can reopen it and review your changes (hopefully a lot faster this time)!

**Pull request closed without merging**